lib/hermann/provider/java_producer.rb

```diff
@@ -43,7 +43,7 @@ def initialize(brokers, opts={})
       # @return +Concurrent::Promise+ Representa a promise to send the
       #   data to the kafka broker.  Upon execution the Promise's status
       #   will be set
-      def push_single(msg, topic, key)
+      def push_single(msg, topic, key, _)
```

Comment:
This makes them the same. I thought that it would be nicer to keep the interface the same. Actually, if we move the `to_java` to the `java_producer`, then we could use the same test for both platforms here: https://github.com/braintree/hermann/blob/stable-partition/spec/producer_spec.rb#L73. If you would rather not change the signature, I can put it back to how it was.